### PR TITLE
feat(repo): add repo permissions for teams

### PR DESCRIPTION
**Pull request by haarchri**
### Description of your changes
uses https://github.com/upbound/up-sdk-go/pull/94 
add repo permission to grant teams permissions

Fixes #

I have:

- [x] Read and followed Upbound's [contribution process](https://git.io/fj2m9).
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR, as appropriate.

### How has this code been tested

```
up repo permission list test123 -a upboundcare                                                
TEAM      REPOSITORY                                 PERMISSION   CREATED                UPDATED
test123   function-conditional-patch-and-transform   write        2024-06-25T06:23:17Z   n/a
```

```
up repo permission revoke test123 function-conditional-patch-and-transform -a upboundcare
Are you sure you want to revoke the permission for team "test123" in repository "function-conditional-patch-and-transform"? This cannot be undone [y/n]: y
Revoking repository permission for team "test123" in repository "function-conditional-patch-and-transform"
Repository permission for team "test123" in repository "function-conditional-patch-and-transform" revoked
```

```
up repo permission grant test123 function-conditional-patch-and-transform write -a upboundcare
Permission "write" granted to team "test123" for repository "function-conditional-patch-and-transform" in account "upboundcare"
```

### CODE REVIEW

**Comment by haarchri** [comment]
needs update after sdk is up2date for https://github.com/upbound/up-sdk-go/pull/94

File: cmd/up/repository/repository.go [hunk 1]
```diff
@@ -77,10 +78,11 @@ func PredictRepos() complete.Predictor {
...
 type Cmd struct {
-	Create createCmd `cmd:"" help:"Create a repository."`
-	Delete deleteCmd `cmd:"" help:"Delete a repository."`
-	List   listCmd   `cmd:"" help:"List repositories for the account."`
-	Get    getCmd    `cmd:"" help:"Get a repository for the account."`
+	Create     createCmd      `cmd:"" help:"Create a repository."`
+	Delete     deleteCmd      `cmd:"" help:"Delete a repository."`
+	List       listCmd        `cmd:"" help:"List repositories for the account."`
+	Get        getCmd         `cmd:"" help:"Get a repository for the account."`
+	Permission permission.Cmd `cmd:"" help:"Assign permissions to a repository for a team in the account."`
```
Comment (turkenh) [inline cmd/up/repository/repository.go:85]:
```suggestion
	Permission permission.Cmd `cmd:"" help:"Manage permissions of a repository for a team in the account."`
```

File: cmd/up/repository/permission/permission.go [hunk 2]
```diff
@@ -0,0 +1,47 @@
+// Copyright 2022 Upbound Inc
```
Comment (turkenh) [inline cmd/up/repository/permission/permission.go:1]:
```suggestion
// Copyright 2024 Upbound Inc
```

File: cmd/up/repository/permission/list.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
+// Copyright 2022 Upbound Inc
```
Comment (turkenh) [inline cmd/up/repository/permission/list.go:1]:
```suggestion
// Copyright 2024 Upbound Inc
```

File: cmd/up/repository/permission/create.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
+// Copyright 2022 Upbound Inc
```
Comment (turkenh) [inline cmd/up/repository/permission/create.go:1]:
```suggestion
// Copyright 2024 Upbound Inc
```

File: cmd/up/repository/permission/create.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+	Permission     string `arg:"" required:"" help:"Permission type (admin, read, write, view)."`
+}
+
+// Validate validates the createCmd struct.
+func (c *createCmd) Validate() error {
+	switch repositorypermission.PermissionType(c.Permission) {
+	case repositorypermission.PermissionAdmin, repositorypermission.PermissionRead, repositorypermission.PermissionWrite, repositorypermission.PermissionView:
+		return nil
+	default:
+		return errors.New("invalid permission type: must be one of [admin, read, write, view]")
```
Comment (turkenh) [inline cmd/up/repository/permission/create.go:44]: nit: would be good to output the provided permission in the error message, e.g. 

`invalid permission type "edit": must be one of [admin, read, write, view]`

File: cmd/up/repository/permission/create.go [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+	}
+}
+
+// Run executes the create command.
+func (c *createCmd) Run(ctx context.Context, p pterm.TextPrinter, ac *accounts.Client, oc *organizations.Client, rpc *repositorypermission.Client, upCtx *upbound.Context) error {
+	a, err := ac.Get(ctx, upCtx.Account)
+	if err != nil {
+		return err
```
Comment (turkenh) [inline cmd/up/repository/permission/create.go:52]: nit: wrap errors.

File: cmd/up/repository/permission/create.go [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+			teamID = t.ID
+			teamFound = true
+			break
+		}
+	}
+	if !teamFound {
+		return fmt.Errorf("could not find team %s in account %s", c.TeamName, upCtx.Account)
+	}
+
+	if err := c.Validate(); err != nil {
```
Comment (turkenh) [inline cmd/up/repository/permission/create.go:77]: This could be moved to the beginning of Run ?

File: cmd/up/repository/permission/delete.go [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *deleteCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if c.Force {
+		return nil
+	}
+
+	confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository permission? This cannot be undone [y/n]", false)
+	if err != nil {
+		return errors.Wrap(err, "error with deletion promt")
```
Comment (adamwg) [inline cmd/up/repository/permission/delete.go:56]:
```suggestion
		return errors.Wrap(err, "error with deletion prompt")
```

File: cmd/up/repository/permission/delete.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+	return nil
+}
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *deleteCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if c.Force {
+		return nil
+	}
+
+	confirm, err := c.prompter.Prompt("Are you sure you want to delete this repository permission? This cannot be undone [y/n]", false)
```
Comment (adamwg) [inline cmd/up/repository/permission/delete.go:54]: I'd suggest we include the team and repo names in the prompt, playing back to the user what they've requested.

File: cmd/up/repository/permission/create.go [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+	if err := rpc.Create(ctx, upCtx.Account, teamID, repositorypermission.CreatePermission{
+		Repository: c.RepositoryName,
+		Permission: repositorypermission.RepositoryPermission{
+			Permission: repositorypermission.PermissionType(c.Permission),
+		},
+	}); err != nil {
+		return errors.Wrap(err, "cannot create permission")
+	}
+	p.Printfln("Permission %s granted to team %s for repository %s in account %s", c.Permission, c.TeamName, c.RepositoryName, upCtx.Account)
```
Comment (adamwg) [inline cmd/up/repository/permission/create.go:89]: Suggest using `%q` for consistency with the other commands.
```suggestion
	p.Printfln("Permission %q granted to team %q for repository %q in account %q", c.Permission, c.TeamName, c.RepositoryName, upCtx.Account)
```

File: cmd/up/repository/permission/revoke.go [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+	return nil
+}
+
+// AfterApply accepts user input by default to confirm the delete operation.
+func (c *revokeCmd) AfterApply(p pterm.TextPrinter, upCtx *upbound.Context) error {
+	if c.Force {
+		return nil
+	}
+
+	confirm, err := c.prompter.Prompt(fmt.Sprintf("Are you sure you want to revoke the permission for team %q in repository %q? This cannot be undone [y/n]", c.TeamName, c.RepositoryName), false)
```
Comment (adamwg) [inline cmd/up/repository/permission/revoke.go:54]: "This cannot be undone" is a little strong :-) The permission can always be re-granted, right?

**Review by adamwg — APPROVED** [review]
lgtm

File: cmd/up/repository/permission/revoke.go (hunk 11 above)
Comment (haarchri) [reply cmd/up/repository/permission/revoke.go:54]: Yeah same for all other resources ;) let's do this rename in a seperate PR

**Pull request merged**